libtransact/src/execution/adapter/static_adapter.rs

```diff
@@ -0,0 +1,619 @@
+                                &context_manager,
+                                context_id,
+                                on_done,
+                            );
+                        }
+                        StaticAdapterCommand::Start(mut execution_registry) => {
+                            register_handlers(&handlers, &mut *execution_registry);
+                        }
+                        StaticAdapterCommand::Stop => {
+                            break;
+                        }
+                    }
+                }
+                true
+            })
+            .map_err(|err| ExecutionAdapterError::GeneralExecutionError(Box::new(err)))?;
+
+        Ok(StaticExecutionAdapter {
+            join_handle,
+            sender,
+        })
+    }
+}
+
+fn execute_transaction(
+    handlers: &[Box<dyn TransactionHandler>],
+    transaction_pair: TransactionPair,
+    context_manager: &ContextManager,
+    context_id: ContextId,
+    on_done: OnDoneCallback,
+) {
+    let family = TransactionFamily::from_pair(&transaction_pair);
+    match handlers.iter().find(|handler| {
+        handler.family_name() == family.family_name()
+            && handler
+                .family_versions()
+                .iter()
+                .any(|v| v == family.family_version())
+    }) {
+        Some(handler) => {
+            let mut static_context = StaticContext::new(context_manager, &context_id);
+
+            match handler.apply(&transaction_pair, &mut static_context) {
+                Ok(_) => on_done(Ok(ExecutionTaskCompletionNotification::Valid(context_id))),
+                Err(ApplyError::InvalidTransaction(error_message)) => {
+                    on_done(Ok(ExecutionTaskCompletionNotification::Invalid(
+                        context_id,
+                        InvalidTransactionResult {
+                            transaction_id: transaction_pair
+                                .transaction()
+                                .header_signature()
+                                .to_owned(),
+                            error_message,
+                            error_data: vec![],
+                        },
+                    )))
+                }
+                Err(err) => on_done(Err(ExecutionAdapterError::GeneralExecutionError(Box::new(
+                    err,
+                )))),
+            }
+        }
+        None => on_done(Err(ExecutionAdapterError::RoutingError(Box::new(
+            transaction_pair,
+        )))),
+    };
+}
+
+fn register_handlers(
+    handlers: &[Box<dyn TransactionHandler>],
+    execution_registry: &mut ExecutionRegistry,
+) {
+    for handler in handlers {
+        for version in handler.family_versions() {
+            execution_registry.register_transaction_family(TransactionFamily::new(
+                handler.family_name().to_owned(),
+                version.clone(),
+            ));
+        }
+    }
+}
+
+impl ExecutionAdapter for StaticExecutionAdapter {
```

Comment:
We should probably propagate up the errors so they can (maybe) be handled by the caller.

Reply:
Fixed in new commits.